src/main/java/org/pac4j/dropwizard/Pac4jFactory.java

```diff
@@ -329,44 +363,84 @@ public void setSessionEnabled(boolean sessionEnabled) {
+            config = (JaxRsConfig) ConfigBuilder.build(configClass, configProperties);
+        } else {
+            config = new JaxRsConfig();
+        }
+        Clients newClients = config.getClients();
+        if (newClients == null) {
+            newClients = new Clients();
+            config.setClients(newClients);
+        }
+        if (newClients.getClients() == null) {
+            newClients.setClients(new ArrayList<>());
+        }
+        config.setClients(newClients);
 
-        config.setClients(client);
         config.setDefaultClients(defaultClients);
 
+        // we can take the clients built from the properties
+        if (clientsProperties != null && clientsProperties.size() > 0) {
+            final PropertiesConfigFactory propertiesConfigFactory = new PropertiesConfigFactory(clientsProperties);
+            final Config propertiesConfig = propertiesConfigFactory.build();
+            config.getClients().getClients().addAll(propertiesConfig.getClients().getClients());
+        }
+        // and the clients directly defined in the YAML file
+        if (clients != null) {
+            config.getClients().getClients().addAll(clients);
+        }
+
         if (callbackUrl != null) {
-            client.setCallbackUrl(callbackUrl);
+            newClients.setCallbackUrl(callbackUrl);
         }
         if (clientNameParameter != null) {
-            client.setClientNameParameter(clientNameParameter);
+            newClients.setClientNameParameter(clientNameParameter);
+        }
+        newClients.setUrlResolver(urlResolver);
+        if (authorizationGenerators != null) {
+            newClients.setAuthorizationGenerators(authorizationGenerators);
         }
-        client.setUrlResolver(urlResolver);
-        client.setAuthorizationGenerators(authorizationGenerators);
-        client.setClients(clients);
 
+        // TODO: to be removed after pac4j#870
         if (defaultClient != null) {
             boolean found = false;
             for (Client c : clients) {
                 if (defaultClient.equals(c.getName())) {
-                    client.setDefaultClient(c);
+                    newClients.setDefaultClient(c);
                     found = true;
                     break;
                 }
             }
             if (!found) {
                 throw new IllegalArgumentException("Client '" + defaultClient
-                        + "' is not one of the configured clients");
+                    + "' is not one of the configured clients");
             }
         }
 
-        config.setSecurityLogic(securityLogic);
-        config.setCallbackLogic(callbackLogic);
-        config.setLogoutLogic(logoutLogic);
-        config.setHttpActionAdapter(httpActionAdapter);
-        config.setAuthorizers(authorizers);
-        config.setMatchers(matchers);
-        config.setProfileManagerFactory(profileManagerFactory);
+        if (securityLogic != null) {
+            config.setSecurityLogic(securityLogic);
```

Comment:
Yes, it can also be set via the `ConfigFactory`. I doubt both will be used at the same time. Given this edge use case, I don't have a strong opinion, I would keep the current behavior: the one explicitly defined in the YML file wins.